ir/src/main/scala/org/scalajs/core/ir/Trees.scala

```diff
@@ -436,6 +436,122 @@ object Trees {
     val tpe = AnyType
   }
 
+  /** Calls a method inherited from the parent class of `cls` on `receiver`.
+   *
+   *  `cls` must be a Scala.js-defined JS class.
+   *
+   *  Given the Scala.js-defined JS class
+   *
+   *  {{{
+   *  @ScalaJSDefined
+   *  class Foo extends Bar
+   *  }}}
+   *
+   *  The node
+   *
+   *  {{{
+   *  JSBrackerSuperCall(ClassType(Foo), receiver, method, args)
+   *  }}}
+   *
+   *  which is printed as
+   *
+   *  {{{
+   *  receiver.Foo::super[method](...args)
+   *  }}}
+   *
+   *  has the following semantics:
+   *
+   *  {{{
+   *  Bar.prototype[method].call(receiver, ...args)
+   *  }}}
+   *
+   *  If this happens to be located in an instance method of `Foo`, *and*
+   *  `receiver` happens to be `This()`, this is equivalent to the ES6
+   *  statement
+   *
+   *  {{{
+   *  super[method](...args)
+   *  }}}
+   */
+  case class JSBracketSuperCall(cls: ClassType, receiver: Tree, method: Tree,
+      args: List[Tree])(implicit val pos: Position) extends Tree {
+    val tpe = AnyType
+  }
+
+  /** Super constructor call in the constructor of a Scala.js-defined JS class.
+   *
+   *  Exactly one such node must appear in the constructor of a
+   *  Scala.js-defined JS class, at the top-level (possibly as a direct child
+   *  of a top-level `Block`).
+   *
+   *  Statements before this node, as well as the `args`, cannot contain any
+   *  `This()` node. Statements after this node can use `This()`.
+   *
+   *  This node's semantics are to call the super constructor, then create
+   *  the fields of the object declared by the current class, initialized to
+   *  the zero of their type.
```

Comment:
I agree a better name would be nice, but I don't see how your suggestion solves the issue. It still fails to highlight the fact that the fields are created at that time. (or it fails to say that the super ctor is called, depending on how you interpret "Basic")

One solution might be to relax a bit the semantics of this node, in a way that could also allow us to take advantage of a possible hypothetical extension of JS in the far future that would create all fields of an object before constructors are called:

change the "This node's semantics" paragraph into:

> After the execution of this node, it is guaranteed that all fields declared in the current class have been created and initialized. Up to that point, accessing any field declared in this class (e.g., through an overridden method called from the super constructor) is undefined behavior.

This way, the node doesn't *provoke* the creation of the fields in its semantics. The field creation semantics is lifted as a guarantee of the structure of the constructor body itself (just like where `This()` can and cannot appear).